File: SHMIP/shmip-hydro_intercomparison-06b3137c7049/input_functions/source/matlab/diurnal_runoff.m
function m=diurnal_runoff(time, relative_amp, steady_state_runoff, basal_melt)
% m=diurnal_runoff(time, relative_amp, steady_state_runoff, basal_melt)
%
% Puts a diurnal signal onto a steady one.  If time is negative, no diurnal signal will be
% added (for spin-up).  If runoff is negative, it is set to zero.  Some small basal melt
% is always present if basal_melt==true.
%
% Input:
%  - time (s)
%  - relative_amp: amp = relative_amp*steady_state_runoff
%  - steady_state_runoff: runoff without diurnal variations
%  - basal_melt: include basal melt if == true

if basal_melt
    basal = steady_runoff(1);
else
    basal = 0;
end

day = 24*60*60;

if time<0
    m = steady_state_runoff;
else
    m = steady_state_runoff*(1 - relative_amp*sin(2*pi*time/day)) ;
end

m(m<0) = 0;

% add basal
m = m + basal;

File: SHMIP/shmip-hydro_intercomparison-06b3137c7049/input_functions/source/python/diurnal_runoff.py
import numpy as np

def DiurnalForcing(time,RelAmp=None):
	'''Runoff=SeasonForcing(surf,time,Tparam)
	Definition of the diurnal input function for experiment C of the SHMIP exercise
	This is only the Moulin input part, distributed source of run A1 has to be added
	
	Input:
	time : are the timestamps of the requested timesteps (one vector)
	RelAmp: is the relative amplitude of the signal as given in SHMIP 

	Output:
	OUtput is the transformed input which has to be used for every moulin.
	'''
	day = 24.*60.*60. #From second to day
	MI = 0.9 #initial noulin input

	if RelAmp is None:
		ra = 0.
	else:
		ra = RelAmp
	

	Input=MI*(1.-ra*np.sin(2.*np.pi*time/day))
	Input[np.where(Input<0)]=0.0

	return Input
